openprescribing/frontend/views/spending_utils.py

```diff
@@ -116,113 +102,51 @@ def _get_names_for_bnf_codes(bnf_codes):
 
 def _get_concession_cost_matrices(min_date, max_date, org_type, org_id):
     """
-    Given a date range (inclusive) and an organisation return a set of matrices
-    detailing all spending affected by price concessions during the period.
-
-    Returns a ConcessionCostMatrices object with the following attributes:
+    Given a date range (inclusive) and an organisation return three
+    LaballedArrays (of the same shape) detailing all spending affected by price
+    concessions during the period.
 
-        bnf_code_offsets: Maps BNF codes which have a price concession to their
-                          row offset in the matrices
-
-            date_offsets: Maps date strings to their column offset in the
-                          matrices
-
-              quantities: Matrix giving, for each BNF code and date, the
+              quantities: LabelledArray giving, for each BNF code and date, the
                           quantity prescribed of that presentation by the
                           specified organisation
 
-            tariff_costs: Matrix giving, for each BNF code and date, the cost
-                          of the above prescriptions at standard Drug Tariff
-                          prices
+            tariff_costs: LabelledArray giving, for each BNF code and date, the
+                          cost of the above prescriptions at standard Drug
+                          Tariff prices
 
-             extra_costs: Matrix giving, for each BNF code and date, the cost
-                          increase due to price concessions
+             extra_costs: LabelledArray giving, for each BNF code and date, the
+                          cost increase due to price concessions
 
     """
-    prices = _get_concession_price_matrices(min_date, max_date)
-    quantities = _get_prescribed_quantity_matrix(
-        prices.bnf_code_offsets, prices.date_offsets, org_type, org_id
-    )
-    return ConcessionCostMatrices(
-        bnf_code_offsets=prices.bnf_code_offsets,
-        date_offsets=prices.date_offsets,
-        quantities=quantities,
-        tariff_costs=prices.tariff_prices * quantities,
-        extra_costs=prices.price_increases * quantities,
-    )
-
-
-def _get_prescribed_quantity_matrix(bnf_code_offsets, date_offsets, org_type, org_id):
+    tariff_prices, price_increases = _get_concession_price_matrices(min_date, max_date)
+    bnf_codes = tariff_prices.row_labels
```

Comment:
Depending on how much the user of the API should know about the implementation of the `LabelledArray`, could this be `tariff_prices.labels('bnf_codes')`?

Reply:
Yes. As with the `sum_columns` discussion above, I think we should give axes names and then dispense with any references to rows or columns.